api/src/main/java/org/hawkular/inventory/base/spi/CanonicalPath.java

```diff
@@ -113,6 +114,32 @@ public CanonicalPath visitUnknown(Object entity, Void parameter) {
         }, null);
     }
 
+    public Entity toEntity() {
```

Comment:
This creates basically invalid entities - like resource without a resource type or metric type without a unit or indeed a resource type with hardcoded version to "1.0".